src/samples/storage_sample_main.cc

```diff
@@ -0,0 +1,150 @@
+#include "googleapis/client/transport/http_transport.h"
+#include "googleapis/client/transport/http_request_batch.h"
+#include "googleapis/client/util/status.h"
+#include "googleapis/strings/strcat.h"
+
+#include "google/storage_api/storage_api.h"  // NOLINT
+
+namespace googleapis {
+
+static const char usage[] =
+    "This is a sample application illustrating the use of the GoogleApis C++\n"
+    "Client. The application makes calls into the Google Cloud Storage API.\n"
+    "The application itself is not particularly useful, rather it just\n"
+    "illustrates how to interact with a live service.\n"
+    "\n"
+    "Usage:\n"
+    "\tstorage_sample <service_account.json> [<cacerts_path>]\n"
+    "\n"
+    "Output:\n"
+    "\n"
+    "A list of Google Cloud Storage bucket links.\n"
+    "\n"
+    "Example:\n"
+    "\t$ bin/storage_sample your-project-id-1cf578086929.json\n"
+    "\thttps://www.googleapis.com/storage/v1/b/1gallon\n"
+    "\thttps://www.googleapis.com/storage/v1/b/2gallon\n"
+    "\thttps://www.googleapis.com/storage/v1/b/5gallon\n"
+    "\tDone!\n"
+    "\n"
+    "\n"
+    "See README.md for more details.\n";
+
+using google_storage_api::StorageService;
+
+using client::HttpTransport;
+using client::HttpTransportLayerConfig;
+using client::OAuth2Credential;
+using client::OAuth2AuthorizationFlow;
+#if HAVE_OPENSSL
+using client::OpenSslCodecFactory;
+#endif
+using client::StatusInvalidArgument;
+
+class StorageSample {
+ public:
+  googleapis::util::Status Startup(int argc, char* argv[]);
+  void Run();
+
+ private:
+
+  OAuth2Credential credential_;
+  std::unique_ptr<StorageService> storage_;
+  std::unique_ptr<OAuth2AuthorizationFlow> flow_;
+  std::unique_ptr<HttpTransportLayerConfig> config_;
+};
+
+/* static */
+util::Status StorageSample::Startup(int argc, char* argv[]) {
+  if ((argc < 2) || (argc > 3)) {
+    return StatusInvalidArgument(usage);
+  }
+
+  // Set up HttpTransportLayer.
+  googleapis::util::Status status;
+  config_.reset(new HttpTransportLayerConfig);
+  client::HttpTransportFactory* factory =
+      new client::CurlHttpTransportFactory(config_.get());
+  config_->ResetDefaultTransportFactory(factory);
+  if (argc > 2) {
+    config_->mutable_default_transport_options()->set_cacerts_path(argv[2]);
+  }
+
+  // Set up OAuth 2.0 flow for a service account.
+  flow_.reset(new client::OAuth2ServiceAccountFlow(
+      config_->NewDefaultTransportOrDie()));
+  string json(std::istreambuf_iterator<char>(std::ifstream(argv[1]).rdbuf()),
+              std::istreambuf_iterator<char>());
+  flow_->InitFromJson(json);
+  flow_->set_default_scopes(StorageService::SCOPES::DEVSTORAGE_READ_ONLY);
+
+  storage_.reset(new StorageService(config_->NewDefaultTransportOrDie()));
+  return status;
+}
+
+
+void StorageSample::Run() {
+  credential_.set_flow(flow_.get());
+  google_storage_api::BucketsResource_ListMethod request(
+      storage_.get(), &credential_, "bookshelf-dotnet");
```

Comment:
(relayed in-person) I don't have access to the bucket and the sample outputs: 

Could not list buckets: Http(401) Unauthorized